File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.java

class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> set = new HashSet<>();
        int left = 0; int right = 0;
        int max = 0;
        while(right < s.length()){
            char letter = s.charAt(right);


            while(set.contains(letter)){
                set.remove(s.charAt(left));
                left++;
            }

            if(right - left + 1 > max){
                max = right -left + 1;
            }

            set.add(letter);
            right++;

        }
        return max;

    }
}

File: Data Structures & Algorithms/binary-search/submission-4.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int i;
        int left = 0, right = nums.size();
        do{
            i = (right + left) /2;
            if(nums[i] > target){
                right = i - 1;
            }else if(nums[i] < target){
                left = i + 1;
            }else{
                return i;
            }
        }while(i != (right + left) /2);
        return -1;
    }
};
